internal/core/service/authentication-service.go

```diff
@@ -0,0 +1,114 @@
+package service
+
+import (
+	"errors"
+	"log"
+
+	repositories "github.com/cyneptic/letsgo-authentication/infrastructure/repository"
+	"github.com/cyneptic/letsgo-authentication/internal/core/entities"
+	"github.com/cyneptic/letsgo-authentication/internal/core/ports"
+	
+	"github.com/google/uuid"
+)
+
+type AuthenticationService struct {
+	db    ports.UserRepositoryContracts
+	redis ports.InMemoryRespositoryContracts
+}
+
+func NewAuthenticationService() *AuthenticationService {
+	db := repositories.NewPostgres()
+	redis := repositories.RedisInit()
+
+	return &AuthenticationService{
+		db:    db,
+		redis: redis,
+	}
+}
+func (u *AuthenticationService) IsUserAlreadyRegisters(newUser entities.User) (bool, error) {
+	res, err := u.db.IsUserAlreadyRegisters(newUser)
+
+	if err != nil {
+		return true, err
+	}
+
+	if res > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+func (u *AuthenticationService) AddUser(newUser entities.User) error {
+
+	isUserAlreadyExist, err := u.IsUserAlreadyRegisters(newUser)
+
+	if err != nil {
+		
+		return err
+	}
+
+	if isUserAlreadyExist == true {
+		err := errors.New("user already registered")
+		return err
+	}
+
+	newUser.Password , _  = HashPassword(newUser.Password)
+
+	err = u.db.AddUser(newUser)
+	return err
+}
+
+func (u *AuthenticationService) LoginHandler(user entities.User) (string, error) {
+	email := user.Email
+	password := user.Password
+
+	foundedUser, err := u.db.LoginHandler(email)
```

Comment:
Login"Handler" insinuates that its a handler. maybe change the name of the functions to Login.